app/models/game.rb

```diff
@@ -28,6 +28,23 @@ def board_state
     board
   end
 
+  def in_check?(color)
+    #checking king
+    king = pieces.find_by(game_id: id, type:'King', color: color)
+    #checking pieces
+    opponent = self.pieces.where(color: !color, captured: false)
```

Comment:
I'd add a method to `game.rb` call `opposite_color` that returns the other color.

So something like

```Ruby
def opposite_color(color)
  return 'white' if color == 'black'
  'black'
end
```